### PR TITLE
Use YAML to define allocations

### PR DESCRIPTION
**Pull request by jasonish**
A request to add a new range led to this...

The YAML definition file allows multiple ranges per ruleset. The
associated generate.py will generate the Markdown for Jekyll, as well
as warn on overlaps.

I also plan to add a way to easily find holes.

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -1,26 +1,25 @@
-# Sid Allocation
+# Sid Allocations
```
Comment (jasonish) [inline index.md]: Typo

File: index.md [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
-| 5000000-5000033     | MalSilo               | [MalSilo](https://malsilo.gitlab.io/feeds/)                                                          |
-| 7724000-7726000     | 3CORESec              | [Sinkholes Ruleset](https://dtection.io/ruleset)                                                     |
-| 10000000-11999999   | Positive Technologies | [PT Security Attack Detection Team ruleset](https://github.com/ptresearch/AttackDetection#sid-range) |
-| 27990000-27999999   | jpgview               | [DOH Rules](https://raw.githubusercontent.com/jpgpi250/piholemanual/master/DOH/DOH.rules)            |
-| 902200000-906200096 | Abuse.ch              | Abuse.ch                                                                                             |
-
+| SID Range | Organization | Note |
+| :-------- | :----------- | :--- |
+| 1000000-1999999 | local | Reserved for Local Use - Put your custom rules in this range to avoid conflicts |
+| 2000000-2103999 | Emerging Threats | [Emerging Threats Open](https://community.emergingthreats.net/) |
```
Comment (jasonish) [inline index.md]: Could justify these.

**Comment by jasonish** [comment]
Sorry about the bad push as well.

File: allocations.yml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      - start: 2620000
+        size: 10000
+
+  emergingthreats:
+    org: Emerging Threats
+    name: Emerging Threats
+    note: "[Emerging Threats](https://doc.emergingthreats.net/bin/view/Main/SidAllocation)"
+    ranges:
+      - start: 2700000
+        size: 100000
```
Comment (jasonish) [inline allocations.yml]: This ET entry had no open or pro.. @zoomequipd? Also, the URL?

Comment (zoomequipd) [reply allocations.yml]: TBD on how this range will get used.

**Comment by zoomequipd** [comment]
I like this idea a lot, sure beats making a table everytime.

File: allocations.yml (hunk 3 above)
Comment (jasonish) [reply allocations.yml]: Ok, just was curious if it was missing.  But not really a needed item anyways. Just these keys need to be unique.

**Pull request closed without merging**